tests/plots/test_line.py

```diff
@@ -0,0 +1,149 @@
+"""Tests for the plots.line module."""
+
+import warnings
+
+import matplotlib.pyplot as plt
+import pandas as pd
+import pytest
+from matplotlib.axes import Axes
+from matplotlib.colors import ListedColormap
+
+from pyretailscience.plots import line
+from pyretailscience.style import graph_utils as gu
+
+
+@pytest.fixture()
+def sample_dataframe():
+    """A sample dataframe for testing."""
+    data = {
+        "x": pd.date_range("2023-01-01", periods=10, freq="D"),
+        "y": range(10, 20),
+        "group": ["A"] * 5 + ["B"] * 5,
+    }
+    return pd.DataFrame(data)
+
+
+@pytest.fixture()
+def _mock_get_base_cmap(mocker):
+    """Mock the get_base_cmap function to return a custom colormap."""
+    cmap = ListedColormap(["#FF0000", "#00FF00", "#0000FF"])
+    mocker.patch("pyretailscience.style.tailwind.get_base_cmap", return_value=cmap)
+
+
+@pytest.fixture()
+def _mock_gu_functions(mocker):
```

Comment:
Is there a reason you're mocking the functions rather than just leaving them as is?